bam_masterdata/excel/excel_to_entities.py

```diff
@@ -0,0 +1,701 @@
+import re
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+import openpyxl
+from openpyxl.worksheet.worksheet import Worksheet

```

Comment:
Ok, I checked the code and it seems `Worksheet` is only used for typing. You can:
```python
import re
from typing import TYPE_CHECKING, Any, Optional, Union

if TYPE_CHECKING:
    from openpyxl.worksheet.worksheet import Worksheet
```

Then, you need to change in all the times a `Worksheet` was typed in a function by `"Worksheet"`, i.e., add the double-quotes, e.g.:
```python
    def get_last_non_empty_row(
        self, sheet: "Worksheet", start_index: int
    ) -> Optional[int]:
```

The thing is that, if a class is only used for typing checks, then it does not need to be imported (and hence added to memory) every time the module is executed. That's why we have this TYPE_CHECKING and we just need to add an if condition + double quotes when typing the specific class.